### PR TITLE
Create the component config for simulator.(#245)

### LINKED ISSUE #245: Create the component config
/kind feature
/kind cleanup
/area simulator

Currently, the simulator is configured via environment variables.
https://github.com/kubernetes-sigs/kube-scheduler-simulator/blob/master/simulator/docs/env-variables.md

I'd like to create the component config to configure the simulator and deprecate those envs in the future.

### PR DESCRIPTION
**Pull request by yanghesong**
Create the component config in simulator.

Signed-off-by: yanghesong <hesong.yang@foxmail.com>

#### What type of PR is this?
/kind feature
/kind cleanup
/area simulator

#### Which issue(s) this PR fixes:

Fixes #245

#### Special notes for your reviewer:
/cc @sanposhiho

### CODE REVIEW

File: simulator/docs/env-variables.md [hunk 1]
```diff
@@ -6,36 +6,6 @@ used to configure kube-scheduler-simulator.
 Please refer [docker-compose.yml](./../../docker-compose.yml) as an example
 use.
 
-`PORT`: (required) This is the port number on which kube-scheduler-simulator
```
Comment (sanposhiho) [inline simulator/docs/env-variables.md:9]: I don't think we should remove them soon. Just marking as deprecated is enough for now. We can remove it later.

File: simulator/config.yml [hunk 2]
```diff
@@ -0,0 +1,45 @@
+# Configuration file for scheduler-simulator
```
Comment (sanposhiho) [inline simulator/config.yml:1]: Sorry, I should've described more in the issue. 

In k8s, component configurations are often managed under version control.

For example
- [KubeSchedulerConfiguration](https://kubernetes.io/docs/reference/scheduling/config/)
- [descheduler's component config](https://github.com/kubernetes-sigs/descheduler/tree/master/pkg/apis/componentconfig).

So, what I expect is something like

```yaml
apiVersion: simulator.kube-scheduler-simulator.x-k8s.io/v1alpha1
kind: SimulatorConfiguration
port: 1212
etcd-url: "http://127.0.0.1:2379"
cors-allowed-origin-list:
  - "http://localhost:3000"
kube-config: ""
kube-api-host: "127.0.0.1"
kube-api-port: 3131
kube-scheduler-config-path: ""
external-import-enabled: false
external-scheduler-enabled: false
```

TBH I'm not so familiar with [code-generator](https://github.com/kubernetes/code-generator), but, I believe you only need to add some annotation on code like `k8s:deepcopy-gen` or `k8s:conversion-gen` etc and code-generator will generate most of codes.

**Comment by sanposhiho** [comment]
Thanks for working on it. Sure, I'll review it (hopefully) this weekend.

**Comment by sanposhiho** [comment]
Sorry, no much time actually. will try to make time to review it this weekend or around the end-of-year holidays.

**Comment by yanghesong** [comment]
> Sorry, no much time actually. will try to make time to review it this weekend or around the end-of-year holidays.

Thanks!

File: simulator/config/config.go [hunk 3]
```diff
@@ -85,73 +94,86 @@ func NewConfig() (*Config, error) {
 	}, nil
 }
```
Comment (sanposhiho) [inline simulator/config/config.go:96]: Could you keep the original way to set configuration via env? We're using this way for a long time and some users may want to use that.

File: simulator/docs/env-variables.md [hunk 4]
```diff
@@ -6,6 +6,9 @@ used to configure kube-scheduler-simulator.
 Please refer [docker-compose.yml](./../../docker-compose.yml) as an example
 use.
 
+## For Simulator
+This part is deprecated. Please refer to the simulator [config.yml](./../config.yml).
```
Comment (sanposhiho) [inline simulator/docs/env-variables.md:10]: Add the note like "We're planning to remove the configuration via environment variables."

Comment (sanposhiho) [reply simulator/docs/env-variables.md:10]: Also, could you add some note on README.md or somewhere new in /docs about config.yml?

**Review by sanposhiho — COMMENTED** [review]
Thanks very much and so sorry for being late review.
Left some comments.

**Comment by sanposhiho** [comment]
@yanghesong Hey, could you rebase & check above comments?

**Comment by yanghesong** [comment]
I'm editing it now. Please wait me for a moment, thanks!



---Original---
From: "Kensei ***@***.***&gt;
Date: Sat, Mar 4, 2023 12:05 PM
To: ***@***.***&gt;;
Cc: "Hesong ***@***.******@***.***&gt;;
Subject: Re: [kubernetes-sigs/kube-scheduler-simulator] Create the componentconfig for simulator.(#245) (PR #249)




 
@yanghesong Hey, could you rebase &amp; check above comments?

**Comment by sanposhiho** [comment]
Yep, thanks. Take your time.

**Comment by yanghesong** [comment]
> Yep, thanks. Take your time.

/cc @sanposhiho  Please help me review this PR again,thanks!

**Review by sanposhiho — APPROVED** [review]
/lgtm
/approve

Great, thanks @yanghesong!

**Pull request merged**